node_api/src/logic/validator.ts

```diff
@@ -138,4 +139,17 @@ export default class Validator {
   private isValidTermCode(data: any): boolean {
     return isObjectHasKeys(data, [IM.CODE, IM.HAS_STATUS, RDFS.LABEL]) && data[IM.CODE] && data[IM.HAS_STATUS] && data[RDFS.LABEL];
   }
+
+  private isValidSummary(data: any): { isValid: boolean; message?: string } {
+    let valid = false;
+    let message: string | undefined = "One or more fields are invalid.";
+    if (isObjectHasKeys(data, [IM.ID, IM.HAS_STATUS, RDFS.LABEL])) {
+      if (this.isValidIri(data).isValid && data[RDFS.LABEL] && data[IM.HAS_STATUS] && data[IM.CODE]) {
```

Comment:
HAS_STATUS check will pass with an empty object. Needs to be refined to check it is an array of TTIriRefs. The hasStatus check should also ensure every TTIriRef is valid (child of IM.HAS_STATUS).